db/backup/backup.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3.7
+
+""" Daily back up function for databases within a local
+MariaDB instance """
+
+from datetime import datetime
+import subprocess
+from google.cloud import storage
+from google.cloud import logging
+import pytz
+
+
+def perform_backup():
+    """Completes a backup of the databases within a local mariadb instance
+    and uploads this backup to GCS."""
+
+    # Logging
+    logging_client = logging.Client()
+    log_name = 'backup_log'
+    logger = logging_client.logger(log_name)
+
+    # Get timestamp
+    utc_now = pytz.utc.localize(datetime.utcnow())
+    timestamp_str = utc_now.strftime('%d_%m_%Y_%H-%M-%S')
+
+    # Export SQL Data
+    try:
+        backup_sql = subprocess.check_output(
+            ['sudo', 'mysqldump', '--lock-tables', '--all-databases']
+        )
+        text = f'Performed mysqldump to pull data {timestamp_str} UTC.'
+        logger.log_text(text, severity='INFO')
+
+    except subprocess.SubprocessError:
+        text = f'Failed to export backup {timestamp_str} UTC.'
+        logger.log_text(text, severity='ERROR')
+        return
+
+    # Write backup to file
+    filepath = f'backups/backup_{timestamp_str}.sql'
+    with open(filepath, 'xb') as f:
```

Comment:
This file isn't ever deleted, and could fill the db hdd pretty quickly.

Reply:
Great point! Have addressed in the latest commit (hopefully)